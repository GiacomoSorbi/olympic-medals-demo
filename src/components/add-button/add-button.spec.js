import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AddButton from './'

describe('AddButton should', () => {
  it('exist', () => {
    expect(AddButton).not.toBeUndefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(<AddButton onClick={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
