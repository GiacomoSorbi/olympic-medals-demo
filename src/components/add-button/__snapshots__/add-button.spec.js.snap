// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton should have a consistent structure 1`] = `
<AddButton
  onClick={[MockFunction]}
>
  <Button
    className="add-button d-flex align-items-center"
    color="primary"
    onClick={[MockFunction]}
    tag="button"
  >
    <button
      aria-label={null}
      className="add-button d-flex align-items-center btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="edit row"
        className="add-icon"
        src="add.svg"
      />
      <div
        className="add-text"
      >
        Add new Entry
      </div>
    </button>
  </Button>
</AddButton>
`;
